File: client-mobile/app/src/main/java/com/example/polyquiz/chat/domain/Message.kt
package com.example.polyquiz.chat.domain

import com.example.polyquiz.auth.domain.UserIdName
import java.util.Date

data class Message(
    val id: String,
    val text: String,
    val authorId: String,
    val authorUsername: String,
    val photoUrl: String,
    val date: Date,
    val userLikes: List<UserIdName>,
    val userLoves: List<UserIdName>,
    val userDislikes: List<UserIdName>
)

File: common/interfaces/message.ts
import { UserIdName } from '../interfaces/user-id-name';

export interface Message {
    id: string;
    text: string;
    authorId: string;
    authorUsername: string;
    photoUrl: string;
    date: Date;
    userLikes: UserIdName[];
    userLoves: UserIdName[];
    userDislikes: UserIdName[];
}
